### ISSUE TITLE
TalkBack is not working correctly on Android TV

### ISSUE THREAD

**Issue by ELCHULI190** [flutter/flutter#170670]
### Steps to reproduce

To reproduce the issue with TalkBack, follow these steps (TalkBack must be activated):
1. Open the app on an Android TV
2. Navigate to the slider
3. Try moving left, right, or in any direction
4. You’ll notice that the app's visual focus does not move, but the TalkBack focus does

This issue is reproducible on Android TV, but it does not happen on Fire TV, where the focus behaves correctly.

### Expected results

Just like on Fire TV, the TalkBack focus correctly matches the app focus.

### Actual results

TalkBack does not rely on the app’s focus, but instead uses its own accessibility focus.
As a result, any logic in the app that depends on the standard focus system does not work properly when TalkBack is active.

This creates issues in several contexts where focus-driven logic is essential for the app to function correctly.

I’ve confirmed that this issue does not occur on Fire TV (Firestick), but it does happen on TVs running Android TV.

### Code sample

<details open><summary>Code sample</summary>

```dart
  ListView.builder(
    controller: rowController,
    scrollDirection: Axis.horizontal,
    shrinkWrap: true,
    itemCount: 5,
    itemBuilder: (context, i) {
      return ListTile(
        title: Text("Tile: $i"),
        onTap: () {
        },
      );
    },
  ),
```
</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.27.4, on macOS 15.5 24F74 darwin-arm64, locale es-ES)
    • Flutter version 3.27.4 on channel stable at /Users/jnunez/Develop/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision d8a9f9a52e (5 months ago), 2025-01-31 16:07:18 -0500
    • Engine revision 82bd5b7209
    • Dart version 3.6.2
    • DevTools version 2.40.3

[✓] Android toolchain - develop for Android devices (Android SDK version 36.0.0)
    • Android SDK at /Users/jnunez/Develop/Android
    • Platform android-35, build-tools 36.0.0
    • ANDROID_SDK_ROOT = /Users/jnunez/Develop/Android/
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 16.2)
    • Xcode at /Users/jnunez/Develop/Xcode.app/Contents/Developer
    • Build 16C5032a
    • CocoaPods version 1.16.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.11+0-17.0.11b1207.24-11852314)

[✓] VS Code (version 1.100.3)
    • VS Code at /Users/jnunez/Develop/Visual Studio Code.app/Contents
    • Flutter extension version 3.112.0
```

</details>

**Comment by darshankawar** [comment]
@ELCHULI190 
Are you using DPAD to navigate on this device ? If so, can you check if https://github.com/flutter/flutter/issues/54984 is similar to your case or not ? If not, try to upgrade to latest stable and re-run your scenario to check if it still persist or not. I am unable to verify due to lack of this device / setup.
